Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -230,6 +230,22 @@ public extension Sequence {
     func first<T: Equatable>(where keyPath: KeyPath<Element, T>, equals value: T) -> Element? {
         return first { $0[keyPath: keyPath] == value }
     }
+
+    /// SwifterSwift: Return a sequence where every value is equal to combine called on the previous element, and the current element.
+    ///
+    ///        [1, 2, 3].scan(0, combine: +) -> [1, 3, 6]
+    ///        ["a", "b", "c"].scan("", combine: +) -> ["a", "ab", "abc"]
+    ///
+    /// - Parameter initial: Initial result
+    /// - Parameter combine: A closure that takes as its arguments the previous value returned by the closure and the next element
+    /// - Returns: A sequence of the intermediate results calling `combine` on successive elements of self and an acuumulator
+    func scan(initial: Element, combine: (Element, Element) throws -> Element) rethrows -> [Element] {
```

Comment:
`scan` isn't an appropriate name. I think something like `cumulate` makes more sense, since the results are cumulative on each call to `combine`.
Actually even `combine` doesn't fit since you're not really combining the values, you're essentially adding them.

If you want it to be like a related function to `reduce` then maybe it should have almost the same interface.
```swift
func cumulate<Result>(
    _ initialResult: Result,
    _ nextPartialResult: (Result, Self.Element) throws -> Result
) rethrows -> [Result]
```

Reply:
I would still go with `scan` because that is common naming for this operation across many languages and also apple's `Combine` framework https://developer.apple.com/documentation/combine/fail/scan(_:_:), so follow the pattern is easier to find and use.

Reply:
I actually based the function from ReactiveX https://reactivex.io/documentation/operators/scan.html

Reply:
Hey, sorry I've been MIA @cHaLkdusT. If `scan` is already a common naming pattern then please change it back :)